e2e-test/workspace_helper.py

```diff
@@ -4,62 +4,62 @@
-
-                return workspace_id, data['name']
-
-            except Exception as e:
-                logging.info(f"ERROR workspace creation for '{workspace_name}' in billing project '{billing_project_name}'. Error: {e}")
-                continue
-            break
-
-    return null
+def create_workspace(billing_project_name, azure_token, rawls_url, workspace_name = ""):
+    for i in range(0,shared_variables.RETRIES):
+        try:
+            # create a new workspace, need to have attributes or api call doesnt work
+            rawls_workspace_api = f"{rawls_url}/api/workspaces"
+            workspace_name = workspace_name if workspace_name else f"e2e-test-api-workspace-{''.join(random.choices(string.ascii_lowercase, k=5))}"
+            logging.info(f"Creating workspace {workspace_name} in {billing_project_name}")
+            request_body= {
+                "namespace": billing_project_name,
+                "name": workspace_name,
+                "attributes": {}
+            }
+            header = {
+                "Authorization": "Bearer " + azure_token,
+                "accept": "application/json"
+            }
+
+            workspace_response = requests.post(url=rawls_workspace_api, json=request_body, headers=header)
+            if workspace_response.status_code != 201:
+                logging.info(f"ERROR workspace creation for '{workspace_name}' in billing project '{billing_project_name}'. Response: {workspace_response}")
+                raise Exception(f"Error creating workspace: {workspace_response.text}")
+
+            # example json that is returned by request:
+            # {
+            #   "attributes": {},
+            #   "authorizationDomain": [],
+            #   "bucketName": "",
+            #   "createdBy": "yulialovesterra@gmail.com",
+            #   "createdDate": "2023-08-03T20:10:59.116Z",
+            #   "googleProject": "",
+            #   "isLocked": False,
+            #   "lastModified": "2023-08-03T20:10:59.116Z",
+            #   "name": "api-workspace-1",
+            #   "namespace": "yuliadub-test2",
+            #   "workspaceId": "ac466322-2325-4f57-895d-fdd6c3f8c7ad",
+            #   "workspaceType": "mc",
+            #   "workspaceVersion": "v2"
+            # }
+            workspace_response_json = workspace_response.json()
+            data = json.loads(json.dumps(workspace_response_json))
+
+            workspace_id = data['workspaceId']
+            logging.info(f"Successfully started workspace creation for '{workspace_name}' in billing project '{billing_project_name}'. Workspace ID returned: {workspace_id}")
+
+            return workspace_id, data['name']
+
+        except Exception as e:
+            logging.info(f"ERROR workspace creation for '{workspace_name}' in billing project '{billing_project_name}'. Error: {e}")
+            continue
+    else:
+        # this will execute if the loop didnt exit abnormally (i.e. with break)
```

Comment:
I got kinda confused reading this comment because it uses double-negative (didn't + abnormal), can you rephrase as a slightly more positive sentence (eg: this will execute if the loop exhausts all its retries without finding a workspace)?

Reply:
Actually...TBH, the error message you raise does a great job explaining, so maybe you don't even need the comment?